hy/compiler.py

```diff
@@ -272,6 +272,10 @@ def __add__(self, other):
         if isinstance(other, ast.excepthandler):
             return self + Result(stmts=[other])
 
+        if isinstance(other, ast.slice):
+            raise TypeError("Can't add %r with slice specific syntax %r" % (
+                self, other))
+
```

Comment:
I get how the nature of `ast.slice` really bit you during the development of this PR, but will this check really be useful for later programmers?